Automattic/pocket-casts-android
# [Paywall Experiment] Track Subscribe button event

## Description
- Track event when tapping on Subscribe button from Profile

> [!NOTE]  
> I found an UI issue when tapping on subscribe button from original layout that I will address it in another PR

Fixes #2625

## Testing Instructions
1.  Log with a free account
2. Go to Profile -> Account
3. Tap on `Subscribe to Plus`
4. ✅ Ensure `plus_promotion_upgrade_button_tapped` event was tracked
5. ✅ Ensure `select_payment_frequency_shown` event was tracked
6. Swipe to Patron
7. Tap on `Subscribe to Patron`
8. ✅ Ensure `plus_promotion_upgrade_button_tapped` event was tracked
9. ✅ Ensure `select_payment_frequency_shown` event was tracked
10. Have `Paywall AB Experiment` feature flag `OFF` and open paywall
11. Tap on `Subscribe to Plus`
12. ✅ Ensure `plus_promotion_upgrade_button_tapped` event was **not** tracked
13. Have `Paywall AB Experiment` feature flag `ON` and open paywall
11. Tap on `Get Pocket Casts Plus` and then tap on `Subscribe to Plus`
12. ✅ Ensure `plus_promotion_upgrade_button_tapped` event was **not** tracked


## Checklist
 
#### I have tested any UI changes...
